File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left, right, mid;
        right = nums.size();
        left = 0;
        mid = right / 2;
        while(left <= right){
            if(nums[mid] == target){
                return mid;
            } else {
                if(nums[mid] < target){
                    left = mid + 1;
                    mid = (left + right) / 2;
                } else if (nums[mid] > target){
                    right = mid - 1;
                    mid = (left + right) / 2;
                }
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) < 1:
            return 0
        
        max_length = 1
        left = 0
        right = 0
        seen = set()

        while right < len(s):
            while s[right] in seen:
                seen.remove(s[left])
                left += 1
            
            if s[right] not in seen:
                seen.add(s[right])
                curr_len = right - left + 1
                max_length = max(curr_len, max_length)
                right += 1
        
        return max_length